src/main/kotlin/app/Configurator.kt

```diff
@@ -4,163 +4,116 @@
+    val default: Config = Config()  // Default values.
+
+    val config: Config
+        get() = default.merge(user).merge(current)
+    var options: Options = Options()
+        set(value) {
+            value.password = PasswordHelper.hashPassword(value.password)
+            field = value
+            current.merge(options)
         }
 
-    val options: Options  // Final options that will be used by app.
-        get() = mergeLevels()
-
     // User directory path.
     val userDir = try {
         System.getProperty("user.home")
     }
     catch (e: SecurityException) { null }
 
     init {
-        if (userDir != null) {
-            config = loadConfig(userDir)
-        }
+        loadFromFile()
     }
 
-    // Merges different levels of options into one options object in the next
-    // order: current, local, user, default.
-    private fun mergeLevels(): Options {
-        val levels = arrayListOf(current, config, default)
-        val merged = Options()
-
-        for (level in levels) {
-            if (merged.username == null) {
-                merged.username = level.username
-            }
-            if (merged.password == null) {
-                merged.password = level.password
-            }
-            if (merged.silent == null) {
-                merged.silent = level.silent
-            }
+    fun loadFromFile() {
+        if (userDir == null) {
+            return
         }
 
-        return merged
-    }
-
-    // Loads config file from specified path.
-    fun loadConfig(path: String): Options {
-        val options = Options()
+        val mapper = ObjectMapper(YAMLFactory()) // Enable YAML parsing.
+        mapper.registerModule(KotlinModule()) // Enable Kotlin support.
 
+        var loadConfig = Config()
         try {
-            val file = File(path, configFileName)
-
-            if (!file.exists() || !file.isFile) {
-                return options  // No configuration file
+            loadConfig = Files.newBufferedReader(Paths.get(userDir,
+                    configFileName)).use {
+                mapper.readValue(it, Config::class.java)
             }
-
-            val config = PropertiesConfiguration(file)
-
-            // Accessing configuration properties. Unknown values should be
-            // null to distinguish them from specified values from other levels.
-            options.username = config.getString(configUsername, null)
-            options.password = config.getString(configPassword, null)
-            options.silent = config.getBoolean(configSilent, null)
-        }
-        catch (e: ConfigurationException) {
-            // Error while loading the properties file.
-        }
-        catch (e: SecurityException) {
+        } catch (e: IOException) {
+            // IO error occurred.
+        } catch (e: SecurityException) {
             // Read access denied.
+        } catch (e: InvalidPathException) {
+            // If the path string cannot be converted to a Path.
```

Comment:
I think this is not a good place to catch it: pass it up.
Also, there is a good practice when all errors in application is wrapped in custom error and passed to main control layer (which is responsible to exit application), where this error should be handled.